lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1005,6 +1007,46 @@ def invocation_step(self, trans, invocation_id, step_id, **kwd):
         )
         return self.__encode_invocation_step(trans, invocation_step)
 
+    def _generate_aws_estimate(self, trans, invocation_id, **kwd):
+        """
+        build AWS cloud estimate
+        """
+        decoded_workflow_invocation_id = self.decode_id(invocation_id)
+        workflow_invocation = self.workflow_manager.get_invocation(trans, decoded_workflow_invocation_id)
+        history = workflow_invocation.history
+
+        aws_dict, metrics, ds_metrics = {}, {}, {}
+        h_contents = self.history_contents_manager.contained(history)
+        for h in h_contents:
+            ds_metrics[h.dataset.id] = {
+                'file_size': int(h.dataset.file_size),
+                'total_size': int(h.dataset.total_size),
+                'uuid': str(h.dataset.uuid)
+            }
+        for i, step in enumerate(workflow_invocation.steps):
+            if step.workflow_step.type == 'tool':
+                for job in step.jobs:
+                    metrics[i] = summarize_job_metrics(trans, job)
```

Comment:
This way you'll only collect the last jobs' metrics or a workflow step since you overwrite `metrics[i]` for each job.

I'd suggest making metrics a list of step metrics, where each metric is a dictionary describing the step and the metric. The structure could look something like
```python
[{'step_id': step.id,
  'step_label': step.label,
  jobs: [{'job_id': job.id, 'job_metrics': metrics, 'job_inputs': [], 'job_outputs': []]
}]
```